rsciio/zspy/_api.py

```diff
@@ -88,7 +90,15 @@ def __init__(self, file, signal, expg, **kwargs):
 
     @staticmethod
     def _get_object_dset(group, data, key, chunks, **kwds):
-        """Creates a Zarr Array object for saving ragged data"""
+        """Creates a Zarr Array object for saving ragged data
+
+        Forces the number of chunks span the array if not a dask array as
+        calculating the chunks for a ragged array is not supported.
+        """
+        if isinstance(data, da.Array):
+            chunks = chunks
+        else:
+            chunks = np.prod(data.shape)
```

Comment:
To remove the codeQL failure:
```suggestion
        if not isinstance(data, da.Array):
            chunks = np.prod(data.shape)
```

Reply:
Yea I'm not sure why I wrote that like that... Maybe coding too early in the morning.